src/task_types/burn_in_tests.rs

```diff
@@ -0,0 +1,334 @@
+                    total_tests: discovered_task.test_list.len(),
+                    task_name: &discovered_task.task_name,
+                    multiversion_name: None,
+                    multiversion_tags: None,
+                };
+                sub_suites.push(self.create_task(&params, index, test, &burn_in_suite_info))
+            }
+        }
+
+        Ok(sub_suites)
+    }
+
+    /// Create an individual execution task for burn_in.
+    ///
+    /// # Arguments
+    ///
+    /// * `params` - Configuration for how suite should be configured.
+    /// * `index` - Index of sub-task in list of sub-tasks being created for the task.
+    /// * `test` - Name of test to generate sub-suite for.
+    /// * `suite_info` - Information about the suite being generated.
+    ///
+    /// # Returns
+    ///
+    /// Shrub task representing the given sub-task.
+    fn create_task(
+        &self,
+        params: &ResmokeGenParams,
+        index: usize,
+        test: &str,
+        suite_info: &BurnInSuiteInfo,
+    ) -> EvgTask {
+        let origin_suite = suite_info.build_origin_suite(&params.suite_name);
+
+        let sub_suite = SubSuite {
+            index: Some(index),
+            name: suite_info.build_display_name(),
+            test_list: vec![test.to_string()],
+            exclude_test_list: None,
+            origin_suite: origin_suite.clone(),
+            mv_exclude_tags: suite_info.multiversion_tags.clone(),
+            is_enterprise: false,
+        };
+
+        self.gen_resmoke_task_service.build_resmoke_sub_task(
+            &sub_suite,
+            suite_info.total_tests,
+            params,
+            Some(origin_suite),
+        )
+    }
+}
+
+impl BurnInService for BurnInServiceImpl {
+    /// Generate the burn_in_tests task for the given build_variant.
+    ///
+    /// # Arguments
+    ///
+    /// * `build_variant` - Name of build variant to generate burn_in_tests for.
+    /// * `task_map` - Map of task definitions in evergreen project configuration.
+    ///
+    /// # Returns
+    ///
+    /// A generated suite to use for generating burn_in_tests.
+    fn generate_burn_in_suite(
+        &self,
+        build_variant: &str,
+        task_map: Arc<HashMap<String, EvgTask>>,
+    ) -> Result<Box<dyn GeneratedSuite>> {
+        let mut sub_suites = vec![];
+        let discovered_tasks = self.burn_in_discovery.discover_tasks(build_variant)?;
+        for discovered_task in discovered_tasks {
+            let task_name = &discovered_task.task_name;
+            if let Some(task_def) = task_map.get(task_name) {
+                sub_suites.extend(self.build_tests_for_task(
+                    &discovered_task,
+                    task_def,
+                    build_variant,
+                )?);
+            }
+        }
+
+        Ok(Box::new(GeneratedResmokeSuite {
+            task_name: "burn_in_tests".to_string(),
+            sub_suites,
+            use_large_distro: false,
```

Comment:
I assume `use_large_distro: false` is also keeping the same as it was before, right?

Reply:
I don't think burn_in has ever supported "use_large_distro", it just generate tasks on the default distro for the build variant. I believe there is a request on the backlog to start supporting large distros, but we would need to think through how best to do that.

Reply:
Got it.